File: rest-service/src/main/java/mb/lib/queue/consts/URL.java
package mb.lib.queue.consts;

import java.net.URI;

import mb.lib.config.Config;

public class URL
{
  private static final Config conf = Config.getInstance();

  private static final String
    JobEndpoint      = "/job/%s",
    QueueEndpoint    = "/queue",
    QueueIdEndpoint  = QueueEndpoint + "/%s",
    QueueJobEndpoint = QueueIdEndpoint + "/job/%d",
    FailedEndpoint   = QueueIdEndpoint + "/failed",
    FailedIDEndpoint = QueueIdEndpoint + "/failed/%d";

  public static URI jobEndpoint() {
    return URI.create(prependHTTP(conf.getQueueHost()))
      .resolve(String.format(JobEndpoint, conf.getJobCategory()));
  }

  public static URI queueJobEndpoint(String queueName, int jobID) {
    return URI.create(prependHTTP(conf.getQueueHost()))
      .resolve(String.format(QueueJobEndpoint, queueName, jobID));
  }

  public static URI failedEndpoint(String queueName) {
    return URI.create(prependHTTP(conf.getQueueHost()))
      .resolve(String.format(FailedEndpoint, queueName));
  }

  public static URI failedIDEndpoint(String queueName, int failID) {
    return URI.create(prependHTTP(conf.getQueueHost()))
      .resolve(String.format(FailedIDEndpoint, queueName, failID));
  }

  public static String prependHTTP(String uri) {
    if (uri.startsWith("http://") || uri.startsWith("https://"))
      return uri;

    return "http://" + uri;
  }
}

File: rest-service/src/main/java/mb/lib/queue/consts/URL.kt
package mb.lib.queue.consts

import mb.api.service.util.Address
import mb.lib.config.Config
import java.net.URI

object URL
{
  private const val JobEndpoint      = "/job/%s"
  private const val QueueEndpoint    = "/queue"
  private const val QueueIdEndpoint  = "$QueueEndpoint/%s"
  private const val QueueJobEndpoint = "$QueueIdEndpoint/job/%d"
  private const val FailedEndpoint   = "$QueueIdEndpoint/failed"
  private const val FailedIDEndpoint = "$QueueIdEndpoint/failed/%d"

  fun jobSubmissionEndpoint(category: String): URI {
    return URI.create(Address.http(Config.queueHost))
      .resolve(String.format(JobEndpoint, category))
  }

  fun queueJobEndpoint(queueName: String, jobID: Int): URI {
    return URI.create(Address.http(Config.queueHost))
      .resolve(String.format(QueueJobEndpoint, queueName, jobID))
  }

  fun failedEndpoint(queueName: String): URI {
    return URI.create(Address.http(Config.queueHost))
      .resolve(String.format(FailedEndpoint, queueName))
  }

  fun failedIDEndpoint(queueName: String, failID: Int): URI {
    return URI.create(Address.http(Config.queueHost))
      .resolve(String.format(FailedIDEndpoint, queueName, failID))
  }

  @Suppress("HttpUrlsUsage")
  fun prependHTTP(uri: String): String {
    if (uri.startsWith("http://") || uri.startsWith("https://"))
      return uri

    return "http://$uri"
  }
}
